include/af/signal.h

```diff
@@ -590,6 +590,23 @@ AFAPI array convolve1(const array& signal, const array& filter, const convMode m
  */
 AFAPI array convolve2(const array& signal, const array& filter, const convMode mode=AF_CONV_DEFAULT, const convDomain domain=AF_CONV_AUTO);
 
+/**
+   C++ Interface for convolution on two dimensional signals
+
+   \snippet test/convolve.cpp ex_image_convolve2
+
+   \param[in]  signal is the input signal
+   \param[in]  filter is the filter that will be used for the convolution operation
+   \param[in]  stride specifies the filter strides along each dimension
+   \param[in]  padding specifies the padding along each dimension
+   \param[in]  dilation specifies the amount to dilate the filter before convolution
+   \return     the convolved array
+
+   \ingroup signal_func_convolve2
+ */
+AFAPI array convolve2(const array& signal, const array& filter,
+                      const dim4 stride, const dim4 padding, const dim4 dilation);
+
```

Comment:
Strided convolution doesn't handle all the combinations that our old convolutions handle from what I saw in the code - I am referring to the combination table in the docs [here](http://arrayfire.org/docs/group__signal__func__convolve2.htm).

What I mean precisely is that this function has a narrow range of inputs compared to earlier convolve2 which should be mentioend in docs

Reply:
We have renamed the function convolve2_v2. This new function will batch using a different approach compared to the first version